libs/cart/src/models/cart-item-input.ts

```diff
@@ -1,34 +1,75 @@
 import { ID } from '@daffodil/core';
 import { DaffProduct } from '@daffodil/product';
 
+/**
+ * An enum for the type of the cart item.
+ */
+// TODO: remove "input" designation
 export enum DaffCartItemInputType {
 	Simple = 'simple',
 	Composite = 'composite',
 	Configurable = 'configurable'
 }
 
+/**
+ * A set of fields specifying how to add a product to the cart.
+ */
 export interface DaffCartItemInput {
+  /**
+   * The type of cart item that correspondes to the product being added to the cart.
+   */
 	type: DaffCartItemInputType;
+  /**
+   * The ID of the product to add to the cart.
+   */
   productId: DaffProduct['id'];
+  /**
+   * How many of the product should be added to the cart.
+   */
 	qty: number;
 }
 
+/**
+ * A more constrained cart item input for simple products.
+ */
 export interface DaffSimpleCartItemInput extends DaffCartItemInput {
 	type: DaffCartItemInputType.Simple;
 }
 
+/**
+ * An cart item input for composite products.
+ */
 export interface DaffCompositeCartItemInput extends DaffCartItemInput {
 	type: DaffCartItemInputType.Composite;
+  /**
+   * The selected composite product options to be added to the cart.
```

Comment:
This sounds like the options themselves are being added directly to the cart. It should be reworded so it's clear the options are for a product that is being added to the cart.

Reply:
changed